cli/azd/docs/featue-toggle.md

```diff
@@ -0,0 +1,61 @@
+# Feature Toggle
+
+The Azure Developer CLI includes a `feature toggle` which allows the cli to ship `in-preview` ( _a.k.a. experimental_ ). This document provides the feature design for it.
+
+## Main requirement
+
+As a customer, I can explicitly ask `azd` to unblock and show any _experimental_ functionality. The features becoming available are not finalized, so they can be changed or removed in future releases.
+
+The strategy to ask `azd` to `toggle` between modes is:
+
+> azd config set experimental on
```

Comment:
I think we also want granular experimental features. This allows users to opt-in to specific features, while not impacting other pieces of the product.

```bash
azd config set experimental on # Turns on all features
azd config set experimental.provision.caching on # Turns on `provision.caching` feature
````

Also, would you recommend instead users did:

`azd config unset experimental`? I think this is better as it avoids `config `pollution. We should treat `off` and `undefined` similarly.